pkgs/tools/x11/xmcp/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, libX11
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xmcp";
+  version = "unstable-2020-10-10";
+
+  src = fetchFromGitHub {
+    owner = "blblapco";
+    repo = "xmcp";
+    rev = "ee56225f1665f9edc04fe5c165809f2fe160a420";
+    sha256 = "sha256-B3YkYrVEg6UJ2ApaVook4N2XvrCboxDMUG5CN9I79Sg=";
+  };
+
+  buildInputs = [ libX11 ];
+
+  installPhase = ''
+    install -Dm755 xmcp $out/bin/xmcp
```

Comment:
Don't forget the `runHook`s

Reply:
> Aren't hooks supposed to just add pre/post directives to the install phase without directly overriding it?

Yes!
And this is the exact reason they should be put in every custom `????phase`s.

Other people can write overlays above Nixpkgs, and decorating custom phases with pre- and post hooks keeps the expression composable.